nixos/modules/services/network-filesystems/samba.nix

```diff
@@ -201,14 +201,10 @@ in
               message   = "If samba.nsswins is enabled, then samba.enableWinbindd must also be enabled";
             }
           ];
-        # Always provide a smb.conf to shut up programs like smbclient and smbspool.
-        environment.etc."samba/smb.conf".source = mkOptionDefault (
-          if cfg.enable then configFile
-          else pkgs.writeText "smb-dummy.conf" "# Samba is disabled."
-        );
       }
 
       (mkIf cfg.enable {
+        environment.etc."samba/smb.conf".source = mkOptionDefault configFile;
```

Comment:
```suggestion
        environment.etc."samba/smb.conf".source = configFile;
```

Reply:
No, what's currently there is correct. You should be able to overwrite etc files in your system configuration.

Reply:
Yeah, with mkForce. Otherwise you never notice, that you are ignoring other nixos options.

Reply:
Thanks for the review! I removed `mkOptionDefault` here.

Reply:
> Otherwise you never notice, that you are ignoring other nixos options.

Fair enough.